integration_tests/src/examples.rs

```diff
@@ -299,3 +468,141 @@ fn get_formated_body(body_str: &str) -> String {
+            } else {
+                Err(anyhow!("video port required for test sample"))
+            }
+        }
+        TestSample::SampleLoop => {
+            if let Some(port) = video_port {
+                ff_stream_video_loop(ip, port, get_asset_path(test_sample).unwrap())
+            } else {
+                Err(anyhow!("video port required for test sample"))
+            }
+        }
+        TestSample::Generic => {
+            if let Some(port) = video_port {
+                ff_stream_testsrc(
+                    ip,
+                    port,
+                    Resolution {
+                        width: 1920,
+                        height: 1080,
+                    },
+                )
+            } else {
+                Err(anyhow!("video port required for generic"))
+            }
+        }
+    }
+}
+
+// receives and plays h264 encoded video and opus encoded audio
+pub fn gst_stream_sample(
+    ip: &str,
+    video_port: Option<u16>,
+    audio_port: Option<u16>,
+    test_sample: TestSample,
+) -> Result<()> {
+    match test_sample {
+        TestSample::BigBuckBunny => gst_stream(
+            ip,
+            video_port,
+            audio_port,
+            path_to_str(get_asset_path(test_sample).unwrap()).as_str(),
+        ),
+        TestSample::ElephantsDream => gst_stream(
+            ip,
+            video_port,
+            audio_port,
+            path_to_str(get_asset_path(test_sample).unwrap()).as_str(),
+        ),
+        TestSample::Sample => gst_stream(
+            ip,
+            video_port,
+            audio_port,
+            path_to_str(get_asset_path(test_sample).unwrap()).as_str(),
+        ),
+        TestSample::SampleLoop => Err(anyhow!(
+            "cannot play sample in loop using gstreamer, try ffmpeg"
+        )),
+        TestSample::Generic => gst_stream_testsrc(ip, video_port, audio_port),
+    }
+}
+
+fn path_to_str(path_buf: PathBuf) -> String {
+    path_buf.into_os_string().into_string().unwrap()
+}
+
+fn get_asset_path(asset: TestSample) -> Option<PathBuf> {
+    match asset {
+        TestSample::BigBuckBunny => Some(get_path(
+            "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
+            concat!(
+                env!("CARGO_MANIFEST_DIR"),
+                "/examples/assets/BigBuckBunny.mp4"
+            ),
+        )),
+        TestSample::ElephantsDream => Some(get_path(
+            "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4",
+            concat!(
+                env!("CARGO_MANIFEST_DIR"),
+                "/examples/assets/ElephantsDream.mp4"
+            ),
+        )),
+        TestSample::Sample | TestSample::SampleLoop => Some(get_path(
+            "https://filesamples.com/samples/video/mp4/sample_1280x720.mp4",
+            concat!(
+                env!("CARGO_MANIFEST_DIR"),
+                "/examples/assets/sample_1280_720.mp4"
+            ),
```

Comment:
```suggestion
            examples_root_dir().join("assets/sample_1280_720.mp4")
```
where `examples_root_dir` returns PathBuf